endpoints/auction.go

```diff
@@ -397,6 +265,14 @@ bidLoop:
 	return finalValidBids[:finalBidCounter]
 }
 
+func isUndimensionedBanner(bid *pbs.PBSBid) bool {
+	return bid.CreativeMediaType == "banner" && (bid.Height == 0 || bid.Width == 0)
+}
+
+func isBidIDAndCodeEqual(adunit *pbs.PBSAdUnit, bid *pbs.PBSBid) bool {
+	return adunit.BidID == bid.BidID && adunit.Code == bid.AdUnitCode
+}
+
```

Comment:
Just a minor comment but I am not exactly sure as to what are we gaining by extracting these conditions in their own separate functions, specially when they are being used in only one spot. IMHO having those conditions inline to the if block as earlier isn't too bad to read for us to extract them.

Reply:
So the `checkForValidBidSize` function does not surpass the 15 conditionals limit. For sonar cloud, every if, form, switch, while and checks evaluated with Boolean operators like `&&` and `||` are accounted for.

Reply:
Ah ok! Yeah, looks like it is complaining about the cognitive complexity for this function and those `&&` and `||`'s account for that but I'd suggest it would be better to actually extract the `for _, adunit := range bidder.AdUnits` loop into a separate function to reduce the cognitive complexity rather than creating the above methods. The reason being, cognitive complexity is:

```
a measure of how hard the control flow of a function is to understand. Functions with high Cognitive Complexity will be difficult to maintain.
```

IMHO, by extracting the above mentioned for loop in it's own function, we will make the control flow of the function much easier to understand and maintain